org-code-javabuilder/lib/src/main/java/org/code/neighborhood/GridSquare.java

```diff
@@ -0,0 +1,119 @@
+package org.code.neighborhood;
+import org.json.simple.JSONObject;
+
+public class GridSquare {
+    private String color;
+    private boolean passable;
+    private int paintCount;
+    private enum SquareType{ WALL, OPEN, START, FINISH, OBSTACLE, STARTANDFINISH, UNKNOWN};
+    private SquareType squareType;
+
+    public GridSquare(JSONObject descriptor) {
+        this.color = "";
+        try {
+            switch(Integer.parseInt(descriptor.get("tileType").toString())) {
+                case 0:
+                    this.squareType = SquareType.WALL;
+                    this.passable = false;
+                    break;
+                case 1:
+                    this.squareType = SquareType.OPEN;
+                    this.passable = true;
+                    break;
+                case 2:
+                    this.squareType = SquareType.START;
+                    this.passable = true;
+                    break;
+                case 3:
+                    this.squareType = SquareType.FINISH;
+                    this.passable = true;
+                    break;
+                case 4:
+                    this.squareType = SquareType.OBSTACLE;
+                    this.passable = false;
+                    break;
+                case 5:
+                    this.squareType = SquareType.STARTANDFINISH;
+                    this.passable = true;
+                    break;
+                default:
+                    this.squareType = SquareType.UNKNOWN;
+                    this.passable = false;
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("\nPlease check the format of your grid.txt file, especially the tileType\n");
```

Comment:
This is a good example -- If the constructor took (string tileType, int value), this code would move to the caller which will likely contain all of the code related to parsing the file format and catching/handling formatting errors.